### PR TITLE
CHANGE (CodeAnalyzer): @W-11326833@: We more strongly enforce PR naming conventions.

### PR DESCRIPTION
**Pull request by jfeingold35**
After this PR is merged, subsequent PRs to `docdev` will be subjected to the following checks:
- A verification that Jekyll can properly build the doc pages.
- A verification that the PR title matches our agreed-upon convention of "type (scope): @W-XXXXX@: Summary".

Additionally, subsequent PRs to `documentation` will be subject to the following checks:
- A verification that Jekyll can properly build the doc pages.
- A verification that the PR title matches our agreed upon convention of "RELEASE: @W-XXXXXX@: Summary".

### CODE REVIEW

File: github-actions/verify-pr-title/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
 	},
 	"dependencies": {
 		"@actions/core": "^1.2.4",
-		"@actions/github": "^4.0.0"
+		"@actions/github": "^5.0.0"
```
Comment (jfeingold35) [inline github-actions/verify-pr-title/package.json:16]: Same with #843, we had to increment the version number in order to bypass this issue: https://github.com/octokit/plugin-rest-endpoint-methods.js/issues/172

File: github-actions/verify-pr-title/src/verifyPrTitle.ts [hunk 2]
```diff
@@ -1,23 +1,44 @@
-// Matches @W-0000@ through @W-99999999@
-const reBugId = /@W-\d{4,8}@/;
+/**
+ * This is a regex segment that matches the accepted options for PR type.
+ * NOTE: This segment allows for exactly one type.
+ */
+const TYPE_SEGMENT = "(NEW|FIX|CHANGE)";
```
Comment (rmohan20) [inline github-actions/verify-pr-title/src/verifyPrTitle.ts:5]: I've noticed we sometimes have "DOC" or "INTERNAL". Can we add these as well?

**Review by rmohan20 — APPROVED** [review]
Added a minor request, but changes look good.

File: .github/workflows/validate-doc-pr.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+jobs:
+  # We need to verify that the PR's title meets our conventions.
+  enforce_pr_title:
+    runs-on: ubuntu-latest
+    name: Enforce PR title convention
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Verify title
+        uses: ./github-actions/verify-pr-title
```
Comment (rmohan20) [inline .github/workflows/validate-doc-pr.yml:14]: @jfeingold35 Just realized after approving PR - do PR titles on doc changes matter? These would never show up on release notes.

File: github-actions/verify-pr-title/src/verifyPrTitle.ts [hunk 4]
```diff
@@ -1,23 +1,60 @@
...
+	"PMD",
+	"RetireJS"
+];
 
+/**
+ * This is a regex segment that matches the accepted options for feature branch PR scope, enclosed
+ * in parentheses.
+ * NOTE: This segment allows for multiple scopes, separated by a pipe (|)
+ */
+const FEATURE_SCOPE_SEGMENT = `\\((${FEATURE_SCOPE_OPTIONS.join("|")})(${FEATURE_SCOPE_OPTIONS.map(s => `\\|${s}`).join("|")})*\\)`;
```
[the remark this answers is not in the document]
Comment (jfeingold35) [reply github-actions/verify-pr-title/src/verifyPrTitle.ts:32]: Should be doable. Stand by for that.

**Review by rmohan20 — APPROVED** [review]

**Comment by teresa-allen-sfdc** [comment]
LGTM

**Pull request merged**